changes/changes.go

```diff
@@ -0,0 +1,612 @@
+package changes
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	golog "github.com/ipfs/go-log"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/tabular"
+	"github.com/qri-io/qri/dsref"
+	qerr "github.com/qri-io/qri/errors"
+	"github.com/qri-io/qri/fsi"
+	"github.com/qri-io/qri/stats"
+)
+
+var (
+	log = golog.Logger("changes")
+	// ErrCompareStats indicates an error with stats comparison
+	ErrCompareStats = errors.New("stats compare error")
+)
+
+// ChangeReportComponent is a generic component used to populate the change report
+type ChangeReportComponent struct {
+	Left  interface{}            `json:"left"`
+	Right interface{}            `json:"right"`
+	Meta  map[string]interface{} `json:"meta,omitempty"`
+}
+
+// ChangeReportDeltaComponent is a subcomponent that can hold
+// delta information between left and right
+type ChangeReportDeltaComponent struct {
+	ChangeReportComponent
+	Title string      `json:"title,omitempty"`
+	Delta interface{} `json:"delta"`
+}
+
+// StatsChangeComponent represents the stats change report
+type StatsChangeComponent struct {
+	Summary *ChangeReportDeltaComponent   `json:"summary"`
+	Columns []*ChangeReportDeltaComponent `json:"columns"`
+}
+
+// ChangeReportResponse is the result of a call to changereport
+type ChangeReportResponse struct {
+	VersionInfo *ChangeReportComponent `json:"version_info,omitempty"`
```

Comment:
I missed the earlier discussion about VersionInfo last week. Is this really the right interface here? VersionInfo's intention (apologies, it's not very well spelled out in the code) is to return a sparse version of a dataset at a specific commit, such that it contains exactly what the `list` operation needs to show minimal information about a dataset. Everything in a VersionInfo, as it is defined by `dsref/version_info`, should be available in the other components below. Or is it just here for the "username", "name", "path", "profileID"? If so, a `dsref.Ref` would be more appropriate.

Perhaps it really is better to use VersionInfo and I'm missing the larger context. But at first glance it seems redundant and over-specified.

Reply:
This is less functional and more utility, @ramfox asked for me to include VersionInfo as it's easier to manage internally on the FE side. No real usecase beyond that as the Commit right bellow it contains all the neccessary info.

Reply:
So we had another discussion on this topic, and for now we're going to keep both in favor of fast iterating and err on the side of sending more than we need till we formalize the structures and essentially re-do some of these structures.